File: c#/Table.cs
using System;
using System.Collections.Generic;

public class Table {
    public List<string> customers;
    public int cost;
    public List<string> order;

    public Table(List<string> _customers) {
        customers = _customers;
        cost = 0;
        order = new List<string>();
    }

    ~Table() {
        System.Console.WriteLine("Instance removed.");
    }

    public void place_order(List<string> orders, Server server, Menu menu) {
        System.Console.WriteLine("place orders to " + server.name);
        server.recieve_order(this, orders);
        this.order = orders;
        for (int i = 0; i < orders.Count; i++){
            this.cost = this.cost + menu.get_food_cost(orders[i]);
        }
    }
    public void get_food(string foods) {
        this.order.Clear();
        System.Console.WriteLine("get food");
    }

    public void eat_food() {
        System.Console.WriteLine("eating food");
    }

    public void pay_money(Server server) {
        server.recieve_money(cost);
        System.Console.WriteLine("pay money" + cost);
    }
}

File: c++/table.cpp
#include "table.h"
#include <string>
#include <iostream>
#include <list>

using namespace std;

Table::Table(list<string> _customers) {
    customers = _customers;
    cost = 0;
    order = list<string>();
}

Table::~Table() {
    cout << "Instance removed." << endl;
}

void Table::place_order(list<string> orders, Server server, Menu menu) {
    cout << " place orders to" << server.name << endl;
    this->order = orders;
    for (string order : orders){
        this->cost = this->cost + menu.get_food_cost(order);
    }
    server.recieve_order(*this, orders);
}

void Table::get_food(string foods) {
    this->order.clear();
    cout << "get food" << endl;
}

void Table::eat_food() {
    cout << "eating food" << endl;
}

void Table::pay_money(Server server) {
    server.recieve_money(this -> cost);
    cout << "pay money" << this -> cost << endl;
}
